contrib/resources/translations/utf-8/fr.txt

```diff
@@ -1818,29 +1875,30 @@ Examples:
 
 .
 :PROGRAM_SERIAL_HELP
-Manage the serial ports.
+Manages the serial ports.
 
-Usage modes:
-  SERIAL /L | /LIST             List the current serial ports.
-  SERIAL PORT# TYPE [settings]  Set the port to the given TYPE.
+Usage:
+  [32;1mserial[0m [37;1m[PORT#][0m                List all or specified serial ports.
+  [32;1mserial[0m [37;1mPORT#[0m [36;1mTYPE[0m [settings]  Set the specified port to the given type.
 
 Where:
-  PORT# : 1, 2, 3, or 4
-  TYPE  : MODEM, NULLMODEM, DIRECTSERIAL, DUMMY, or DISABLED
+  [37;1mPORT#[0m The port number: [37;1m1[0m, [37;1m2[0m, [37;1m3[0m, or [37;1m4[0m
+  [36;1mTYPE[0m  The port type: [36;1mMODEM[0m, [36;1mNULLMODEM[0m, [36;1mDIRECTSERIAL[0m, [36;1mDUMMY[0m, or [36;1mDISABLED[0m
 
-Optional settings for each TYPE:
-  For MODEM        : IRQ, LISTENPORT, SOCK
-  For NULLMODEM    : IRQ, SERVER, RXDELAY, TXDELAY, TELNET,
+Notes:
+  Optional settings for each [36;1mTYPE[0m:
+  For [36;1mMODEM[0m        : IRQ, LISTENPORT, SOCK
+  For [36;1mNULLMODEM[0m    : IRQ, SERVER, RXDELAY, TXDELAY, TELNET,
                      USEDTR, TRANSPARENT, PORT, INHSOCKET, SOCK
-  For DIRECTSERIAL : IRQ, REALPORT (required), RXDELAY
-  For DUMMY        : IRQ
+  For [36;1mDIRECTSERIAL[0m : IRQ, REALPORT (required), RXDELAY
+  For [36;1mDUMMY[0m        : IRQ
 
-Examples using port 1, or COM1 in DOS:
-  SERIAL /list                                 : List the current serial ports
-  SERIAL 1 NULLMODEM PORT:1250                 : Listen on TCP:1250 as server
-  SERIAL 1 NULLMODEM SERVER:10.0.0.6 PORT:1250 : Connect to TCP:1250 as client
-  SERIAL 1 MODEM LISTENPORT:5000 SOCK:1        : Listen on UDP:5000 as server
-  SERIAL 1 DIRECTSERIAL REALPORT:ttyUSB0       : Use a physical port on Linux
+Examples:
+  [32;1mSERIAL[0m                                       : List the current serial ports
+  [32;1mSERIAL[0m [37;1m1[0m [36;1mNULLMODEM[0m PORT:1250                 : Listen on TCP:1250 as server
+  [32;1mSERIAL[0m [37;1m2[0m [36;1mNULLMODEM[0m SERVER:10.0.0.6 PORT:1250 : Connect to TCP:1250 as client
+  [32;1mSERIAL[0m [37;1m3[0m [36;1mMODEM[0m LISTENPORT:5000 SOCK:1        : Listen on UDP:5000 as server
+  [32;1mSERIAL[0m [37;1m4[0m [36;1mDIRECTSERIAL[0m REALPORT:ttyUSB0       : Use a physical port on Linux
```

Comment:
@shermp ,

On the source side, I'm seeing the tags being used:

``` c++
        MSG_Add("PROGRAM_SERIAL_HELP",
                "Manages the serial ports.\n"
                "\n"
                "Usage:\n"
                "  [color=green]serial[reset] [color=white][PORT#][reset]                List all or specified serial ports.\n"
                "  [color=green]serial[reset] [color=white]PORT#[reset] [color=cyan]TYPE[reset] [settings]  Set the specified port to the given type.\n"
                "\n"
                "Where:\n"
                "  [color=white]PORT#[reset] The port number: [color=white]1[reset], [color=white]2[reset], [color=white]3[reset], or [color=white]4[reset]\n"
                "  [color=cyan]TYPE[reset]  The port type: [color=cyan]MODEM[reset], [color=cyan]NULLMODEM[reset], [color=cyan]DIRECTSERIAL[reset], [color=cyan]DUMMY[reset], >
                "\n"
                "Notes:\n"
                "  Optional settings for each [color=cyan]TYPE[reset]:\n"
                "  For [color=cyan]MODEM[reset]        : IRQ, LISTENPORT, SOCK\n"
                "  For [color=cyan]NULLMODEM[reset]    : IRQ, SERVER, RXDELAY, TXDELAY, TELNET,\n"
                "                     USEDTR, TRANSPARENT, PORT, INHSOCKET, SOCK\n"
                "  For [color=cyan]DIRECTSERIAL[reset] : IRQ, REALPORT (required), RXDELAY\n"
                "  For [color=cyan]DUMMY[reset]        : IRQ\n"
                "\n"
                "Examples:\n"
                "  [color=green]SERIAL[reset]                                       : List the current serial ports\n"
                "  [color=green]SERIAL[reset] [color=white]1[reset] [color=cyan]NULLMODEM[reset] PORT:1250                 : Listen on TCP:1250 as server\n"
                "  [color=green]SERIAL[reset] [color=white]2[reset] [color=cyan]NULLMODEM[reset] SERVER:10.0.0.6 PORT:1250 : Connect to TCP:1250 as client\n"
                "  [color=green]SERIAL[reset] [color=white]3[reset] [color=cyan]MODEM[reset] LISTENPORT:5000 SOCK:1        : Listen on UDP:5000 as server\n"
                "  [color=green]SERIAL[reset] [color=white]4[reset] [color=cyan]DIRECTSERIAL[reset] REALPORT:ttyUSB0       : Use a physical port on Linux\n");
```

But then during write-out, the tags are lost/dropped, making it much more precarious for translators.

I guess we need a way to get to the original marked-up version of the messages when doing these string dumps (and not the ANSI-rendered versions).

Reply:
Oh dear, I hadn't accounted for that...

The conversion happens before messages are saved to the vector (to avoid having to convert multiple times).

I suppose we could convert at MSG_Get, and take the conversion hit. Or save a parallel vector of original strings.

Reply:
Resolved under https://github.com/dosbox-staging/dosbox-staging/pull/1647